src/modules/keyboardmanager/ui/EditKeyboardWindow.cpp

```diff
@@ -214,11 +274,29 @@ void createEditKeyboardWindow(HINSTANCE hInst, KeyboardManagerState& keyboardMan
 
     // Main Header Apply button
     Button applyButton;
-    applyButton.Content(winrt::box_value(L"Apply"));
-    header.SetAlignRightWithPanel(applyButton, true);
-    header.SetLeftOf(cancelButton, applyButton);
+    applyButton.Content(winrt::box_value(L"OK"));
+    applyButton.Background(winrt::Windows::UI::Xaml::Media::SolidColorBrush{ WindowsColors::get_accent_color() });
+    applyButton.Foreground(winrt::Windows::UI::Xaml::Media::SolidColorBrush{ winrt::Windows::UI::Colors::White() });
+    header.SetAlignRightWithPanel(cancelButton, true);
+    header.SetLeftOf(applyButton, cancelButton);
     applyButton.Flyout(applyFlyout);
-    applyButton.Click([&](winrt::Windows::Foundation::IInspectable const& sender, RoutedEventArgs const&) {
+
+    auto ApplyRemappings = [&keyboardManagerState, _hWndEditKeyboardWindow, keyRemapTable, settingsMessage, applyButton]() {
+        KeyboardManagerHelper::ErrorType isSuccess = KeyboardManagerHelper::ErrorType::NoError;
+        // Save the updated shortcuts remaps to file.
+        bool saveResult = keyboardManagerState.SaveConfigToFile();
+        if (!saveResult)
+        {
+            isSuccess = KeyboardManagerHelper::ErrorType::SaveFailed;
+        }
+        settingsMessage.Text(KeyboardManagerHelper::GetErrorMessage(isSuccess));
+        if (isSuccess == KeyboardManagerHelper::ErrorType::NoError)
+        {
+            PostMessage(_hWndEditKeyboardWindow, WM_CLOSE, 0, 0);
+        }
+    };
+
+    applyButton.Click([&keyboardManagerState, keyRemapTable, header, applyFlyout, ApplyRemappings, applyButton, settingsMessage](winrt::Windows::Foundation::IInspectable const& sender, RoutedEventArgs const&) {
         KeyboardManagerHelper::ErrorType isSuccess = KeyboardManagerHelper::ErrorType::NoError;
         // Clear existing Key Remaps
         keyboardManagerState.ClearSingleKeyRemaps();
```

Comment:
We need to move this logic of clearing single key remaps and adding remaps to the keyboardManagerState variable to the ApplyRemappings function. When we save settings we currently do not read from them and on C++ side the reading of settings only happens when PowerToys.exe is opened, after that it stores the settings in memory and persists it whenever required.

Without moving this if you add a remap and get the orphan warning and then press Cancel and then close the window, the remappings would be applied but would not appear in the JSON file. The quickest way to change this is to make a clean copy of this keyboardManagerState variable and add everything there to find if there any errors or if orphaning happens and only once we get final confirmation i.e. ApplyRemappings is called we do the same on that variable